src/mpid/ch4/shm/ipc/src/ipc_win.c

```diff
@@ -138,7 +124,8 @@ int MPIDI_IPC_mpi_win_create_hook(MPIR_Win * win)
     /* Skip shared_table initialization if all local processes are zero size. */
     for (i = 0; i < shm_comm_ptr->local_size; i++) {
```

Comment:
We need to update the comment

Reply:
I don't like the way of using threshold -1 and let XPMEM return a mem_handle even when size is zero. It might be cleaner to (1) return `MPIDI_SHM_IPC_TYPE__NONE` at `get_mem_attr` which means the handle is invalid and remote process should not open it, (2) here checks both `ipc_shared_table [i].ipc_type ` and ` ipc_shared_table [i].size` for each process in the window.

Besides, the check here seems incorrect. Should `attr.ipc_type == MPIDI_SHM_IPC_TYPE__NONE` be `ipc_shared_table[i].ipc_type == MPIDI_SHM_IPC_TYPE__NONE`? It checks if any process's non-zero buffer cannot be supported by IPC. If so, no one can do SHM-based RMA.

I think we don't have to force all processes to have the same IPC type (e.g., one can be GPU buffer and the other can be XPMEM buffer as long as we can get both memory handles.)

In summary, I think the check here can be changed to:
```
    /* Skip shared_table initialization if:
     *  1. All local processes are zero size, or
     *  2. Any process has a non-zero buffer but the IPC type is NONE (i.e. no submodule supports the buffer type) */
    for (i = 0; i < shm_comm_ptr->local_size; i++) {
        total_shm_size += ipc_shared_table[i].size;
        if (ipc_shared_table[i].size > 0 && ipc_shared_table[i].ipc_type == MPIDI_SHM_IPC_TYPE__NONE) {
            total_shm_size = 0;
            break;
        }
    }
```